FCP adapter uses clip item rate instead of source rate

bashesenaxis opened the pull request:
addresses #174 
- uses clip item rate instead of source rate
- edited clipitem in premiere_example.xml that has different rate compared to its linked file
- fixed bug in timeline_widget, transitions don`t have trimmed_range_in_parent method
- fixed bug in timeline_widget, left over rename from stack to composition

ssteinbach commented on opentimelineview/timeline_widget.py:
```diff
@@ -255,8 +255,8 @@ def __init__(self, track, *args, **kwargs):
-        for item in self.track:
-            timeline_range = item.trimmed_range_in_parent()
+        for n, item in enumerate(self.track):
+            timeline_range = self.track.trimmed_range_of_child_at_index(n)
```
@jminor this is another case where we might want to consider folding transitions back into the regular `item` pattern.

jminor replied:
I agree.

jminor replied:
That shouldn't block this change, we can do that later.

ssteinbach replied:
Agreed.

bashesenaxis commented on opentimelineio/adapters/fcp_xml.py:
```diff
@@ -399,42 +418,48 @@ def _parse_track(track_e, kind, rate, element_map):
 
         if track_item.tag == 'clipitem':
-            if start == -1:
+            if start.value == -1:
```
-1 is actually the values that you`ll encounter in an XML file. 
If start or end is -1, then it means that item is preceded or succeeded by a transition. As such, you`ll need to query neighboring items for the correct values.

jminor replied:
Ok, got it.

bashesenaxis commented:
I have rebased my changes and addressed the issue that was causing the unit test to fail. Item rate is now maintained through the read / write process.

bashesenaxis commented:
Cool :)
I`ve got no more changes for this.
